cpp/oneapi/dal/algo/svm/backend/model_impl.hpp

```diff
@@ -37,10 +40,39 @@ class detail::v1::model_impl : public base {
     double second_class_label;
     std::int64_t class_count = 2;
 
+    model_impl() = default;
+    model_impl(const model_impl&) = delete;
+    model_impl& operator=(const model_impl&) = delete;
+    model_impl(backend::model_interop* interop) : interop_(interop) {}
+
+    ~model_impl() {
+        delete interop_;
+    }
+
     backend::model_interop* get_interop() const {
         return interop_;
     }
 
+    void serialize(dal::detail::output_archive& ar) const override {
+        ar(support_vectors, coeffs, bias, biases);
+
+        if constexpr (std::is_same_v<Task, task::classification>) {
+            ar(first_class_label, second_class_label, class_count);
+        }
+
+        dal::detail::serialize_polymorphic(interop_, ar);
+    }
+
+    void deserialize(dal::detail::input_archive& ar) override {
```

Comment:
Just checking: do we allow to deserialize to non-empty model object? Will it flush the previous state completely and replaced by the state from the archive?

Reply:
Yes